package/kedro_viz/integrations/kedro/data_loader.py

```diff
@@ -109,12 +109,81 @@ def load_data(
 
         context = session.load_context()
         session_store = session._store
+
+        # Update the DataCatalog class for a custom implementation
+        # to handle kedro.io.core.DatasetError from
+        # `settings.DATA_CATALOG_CLASS.from_config`
+        if is_lite:
+            settings.DATA_CATALOG_CLASS = DataCatalogLite
+
         catalog = context.catalog
 
         # Pipelines is a lazy dict-like object, so we force it to populate here
         # in case user doesn't have an active session down the line when it's first accessed.
         # Useful for users who have `get_current_session` in their `register_pipelines()`.
         pipelines_dict = dict(pipelines)
         stats_dict = _get_dataset_stats(project_path)
-
     return catalog, pipelines_dict, session_store, stats_dict
+
+
+def load_data(
```

Comment:
does it make sense to break down the function like this 

public method `load_data()` which called private `_load_data_lite` if `is_lite` else `_load_data`

Reply:
Yes I initially did that when I created a dummy session, had a different config loader etc. 

But now we have most of the parts like session, context etc same in both normal and lite version. Also since lite version supports partial rendering (i.e., if some dependencies are present we stick to that), the only change we did is to initialize DatCatalogLite and patch the mocked dependencies. 

I felt this is a good way to breakdown to reuse session creation and other necessary objects